honeybee_vtk/config.py

```diff
@@ -5,200 +5,223 @@
 import pathlib
 import warnings
 
-from typing import List, Optional
-from pydantic import BaseModel, validator, Field
-
-from .types import model_dataset_names
-from .legend_parameter import Colors, Font, LabelFormat, Orientation
+from typing import List, Union
+from pydantic import BaseModel, validator, Field, constr
+from .types import DataSetNames
+from .legend_parameter import ColorSet, Text, DecimalCount, Orientation
 from .model import Model
 from ._helper import get_line_count, get_min_max
 from .vtkjs.schema import DisplayMode
 from .scene import Scene
 
 
-class FontConfig(BaseModel):
-    """Config for the fonts to be used in a legend."""
+class Autocalculate(BaseModel):
+
+    type: constr(regex='^Autocalculate$') = 'Autocalculate'
+
+
+class TextConfig(BaseModel):
+    """Config for the text to be used in a legend.
+
+    This object applies to text for legend title and legend labels as well.
+    """
+
+    class Config:
+        validate_all = True
+        validate_assignment = True
 
     color: List[int] = Field(
         [0, 0, 0],
         description='An array of three integer values representing R, G, and B values'
-        ' for the color of fonts.'
+        ' for the color of text. Values from 0 to 255 are accepted.'
     )
 
     size: int = Field(
         30,
-        description='Size of fonts in points.'
+        description='Text size in points.'
```

Comment:
Can this be negative? Maybe you should include `gt=0` if it can't be negative.

Reply:
Complied